kolibri/core/discovery/models.py

```diff
@@ -117,7 +120,16 @@ class Meta:
         proxy = True
 
     def save(self, *args, **kwargs):
+        from kolibri.core.discovery.utils.network.connections import (
+            capture_network_state,
+        )
+        from kolibri.core.discovery.utils.network.client import NetworkClient
+
         self.dynamic = False
+
+        client = NetworkClient.build_from_network_location(self)
+        self.is_local = capture_network_state(self, client)
```

Comment:
@akolson This wasn't what I meant by my previous requested change, but I can guess why you went about this and it's good motive. 

Of course, we need to make a request in order to determine the IP address, and therefore the value for this new boolean field. Although, doing this within the model's save method isn't acceptable because it could generate HTTP requests and block the save. It could also throw errors if the location isn't contactable. The circular dependency issues are red flags that should alert you that something isn't meshing right. The relationship between the two operate in reverse-- the function `capture_network_state` is called with the `NetworkLocation` object and it does so elsewhere in the code where we have dedicated processing for updating network location data such as this.

So your changes to `capture_network_state` should set the value on the `NetworkLocation` object inside of it, and the function shouldn't return anything. Doing so will eliminate these circular dependencies. You'll also likely need to update the tests for `capture_network_state`

Reply:
Thanks Blaine for the clarification. Changes have been implemented as requested